frontend/packages/console-app/src/components/cloud-shell/AutoResizeTerminal.tsx

```diff
@@ -0,0 +1,78 @@
+import * as React from 'react';
+import { Terminal as XTerminal } from 'xterm';
+import * as fit from 'xterm/lib/addons/fit/fit';
+
+XTerminal.applyAddon(fit);
+
+type TerminalProps = {
+  onData: (data: string) => void;
+  className?: string;
+  shouldResize?: boolean;
+};
+
+export class Terminal extends React.Component<TerminalProps> {
+  private terminalRef;
+
+  private terminal;
+
+  private resizeObserver;
+
+  constructor(props) {
+    super(props);
+    this.terminalRef = React.createRef();
+    const options = {
+      fontFamily: 'monospace',
+      fontSize: 16,
+      cursorBlink: false,
+      cols: 80,
+      rows: 25,
+      padding: 30,
```

Comment:
I think `4px` max is sufficient padding. It looks to align with UX as well. Furthermore we don't want to add more than this amount to the top and bottom.